core/modules/counter.cc

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright(c) 2019 Intel Corporation
+ */
+#include "counter.h"
+/* for GetDesc() */
+#include "utils/format.h"
+/* for endian functions */
+#include <arpa/inet.h>
+/*----------------------------------------------------------------------------------*/
+const Commands Counter::cmds = {
+    {"add", "CounterAddArg", MODULE_CMD_FUNC(&Counter::AddCounter),
+     Command::THREAD_SAFE},
+    {"remove", "CounterRemoveArg", MODULE_CMD_FUNC(&Counter::RemoveCounter),
+     Command::THREAD_SAFE}};
+/*----------------------------------------------------------------------------------*/
+CommandResponse Counter::AddCounter(const bess::pb::CounterAddArg &arg) {
+  uint32_t ctr_id = arg.ctr_id();
+
+  /* check_exist is still here for over-protection */
+  if (counters.find(ctr_id) == counters.end()) {
+    SessionStats s = {.pkt_count = 0, .byte_count = 0};
+    counters.insert(std::pair<uint32_t, SessionStats>(ctr_id, s));
+  } else
+    return CommandFailure(EINVAL, "Unable to add ctr");
+  return CommandSuccess();
+}
+/*----------------------------------------------------------------------------------*/
+CommandResponse Counter::RemoveCounter(const bess::pb::CounterRemoveArg &arg) {
+  uint32_t ctr_id = arg.ctr_id();
+
+  /* check_exist is still here for over-protection */
+  if (counters.find(ctr_id) != counters.end()) {
+    std::cerr << this->name() << "[" << ctr_id
+              << "]: " << counters[ctr_id].pkt_count << ", "
+              << counters[ctr_id].byte_count << std::endl;
```

Comment:
DLOG

Reply:
This will be converted to `DLOG` once counter module is finalized.